packages/core/test/acceptance/di_spec.ts

```diff
@@ -1802,6 +1802,38 @@ describe('di', () => {
     });
   });
 
+  describe('attribute tokens', () => {
+    it('should be able to provide an attribute token', () => {
+      const TOKEN = new InjectionToken<string>('Some token');
+      function factory(token: string): string {
+        return token + ' with factory';
+      }
+      @Component({
+        selector: 'my-comp',
+        template: '...',
+        providers: [{
+          provide: TOKEN,
+          deps: [[new Attribute('token')]],
+          useFactory: factory,
+        }]
+      })
+      class MyComp {
+        constructor(@Inject(TOKEN) readonly token: string) {}
+      }
+
+      @Component({template: `<my-comp token='token'></my-comp>`})
```

Comment:
```suggestion
      @Component({template: `<my-comp token='token_value'></my-comp>`})
```
As is right now the code could inject the attribute name instead of attribute value and the test would pass. Please make these different.